src/posthog-core.ts

```diff
@@ -1456,7 +1463,7 @@ export class PostHog {
      */
 
     get_session_id(): string {
-        return this.sessionManager.checkAndGetSessionAndWindowId(true).sessionId
+        return this.sessionManager?.checkAndGetSessionAndWindowId(true).sessionId ?? ''
```

Comment:
Why not throw instead of returning a bogus value?

Reply:
Because the main use case all of this solves is people calling Posthog methods without having called init. I.e. so they can still pepper posthog all over their app but only call init if it's running in production for example. 

If we throw (which is essentially what we currently do) then their apps crash and they have to do a bunch of workarounds to ensure they first check if Posthog is initialised. 

We're stuck with the original design of the SDK here so...

Reply:
Makes sense, let's just sprinkle in some `console.warnings()` so that people who miss `posthog.init()` know what's happening

Reply:
Do we want that though? Could mean 100s of warning console logs with no way to turn it off... Could imagine that will be annoying...

Reply:
True. One time per instance maybe? E.g. `if (this.instance.guardAgainstOperationBeforeInit()`) return`, where `guardAgainstOperationBeforeInit` would only ever log once